### ISSUE TITLE
Socket Invalid argument error when connecting with proxy on Windows 10 and Python 3

### ISSUE THREAD

**Issue by devmonkey22** [websocket-client/websocket-client#571]
Similar to #426, on Windows 10, getaddrinfo returns socktype=0 when behind a proxy but no proxy information is set (I don't need/want it set because I'm accessing intranet resources).
This causes socket.socket() on python 3.7 to fail with Invalid argument.  Python 2 accepted 0 without error.

```
  File "/usr/lib/python3.7/site-packages/websocket/_core.py", line 514, in create_connection
    websock.connect(url, **options)
  File "/usr/lib/python3.7/site-packages/websocket/_core.py", line 223, in connect
    options.pop('socket', None))
  File "/usr/lib/python3.7/site-packages/websocket/_http.py", line 120, in connect
    sock = _open_socket(addrinfo_list, options.sockopt, options.timeout)
  File "/usr/lib/python3.7/site-packages/websocket/_http.py", line 161, in _open_socket
    sock = socket.socket(family, socktype, proto)
  File "/usr/lib/python3.7/socket.py", line 151, in __init__
    _socket.socket.__init__(self, family, type, proto, fileno)
  OSError: [Errno 22] Invalid argument
```

Is there any reason that https://github.com/websocket-client/websocket-client/blob/master/websocket/_http.py#L142 shouldn't also pass socket.SOCK_STREAM explicitly like it does on line 101?  Temporarily changing my local environment with this change resolves the error.

Thoughts?

Thanks!
Mike

**Comment by pawelmat** [comment]
Can you please make a release with this fix so that pip would automatically pick this up? I had the same problem on a certain configuration with Python 2.7 and Cygwin on Win7 and just spent the last 2 hours tracking it down to exactly this problem+fix.

**Comment by susundberg** [comment]
For the record, i just had the same issue with websocket_client 0.53. Workaround is changing the _http.py line as in https://github.com/websocket-client/websocket-client/pull/573/files.

But at least in my case (errror 22) issue was also solved by installing 0.57 from pip (-> so this ticket should be closed?)

**Comment by engn33r** [comment]
As mentioned by previous comments, PR #573 addressed this issue and is merged into the current codebase, so this should be closed.
